### ISSUE TITLE
bucket policy: add condition `aws:SecureTransport`

### ISSUE THREAD

**Issue by kuenishi** [basho/riak_cs#380]
One of the lost part of bucket policy that is required CC: #356

**Comment by kuenishi** [comment]
```
{
    "Id": "Policy1355283297687",
    "Statement": [
        {
            "Sid": "Stmt1355283289hage",
            "Action": [
                "s3:PutObject", "s3:DeleteObject", "s3:GetObject"
            ],
            "Effect": "Deny",
            "Resource": "arn:aws:s3:::kuenishi.test/*",
            "Principal": { "AWS": ["*"] },
            "Condition": {
                "Bool": { "aws:SecureTransport" : false }
            }
        }
    ]
}
```

s3 policy accepts json like this. the `"Bool": { "aws:SecureTransport" : false }` - `false` is supposed to be boolean, but aws s3 accepts any json objects. sigh.
Should it be really S3 compatible?
- plan A "compatibility": accept anything as S3 does
- plan B "justice": accept only true/false as the document indicates

**Comment by kuenishi** [comment]
I prefer functional style of plan B.

**Comment by kuenishi** [comment]
Hooray!!! This is how S3 behaves

```
%% "Bool": { "aws:SecureTransport" : true,
%%           "aws:SecureTransport" : false } is recognized as false
%%
%% "Bool": { "aws:SecureTransport" : false,
%%           "aws:SecureTransport" : true } is recognized as true
%% "true" => true
%% "TRUE" => true
%% "True" => true
%% "tRUe" => true
%% "true   " => false
%% "trueboo" => false
%% "true boo" => false
%% "false" => false
%% "falsetrue" => false
%% "0" => false
%% "1" => false
%% "oopaa" => false
%% [] => []
%% ["true"] => true
%% ["false", "true"] => [false,true]
```

**Comment by kuenishi** [comment]
merged #401 .
